File: fetch-api/connectors/ml/src/routes/internal.py
from connectors.ml.settings import settings
from fastapi import APIRouter


router = APIRouter()


@router.get('/health', tags=['internal'], summary='Health check')
def health() -> dict:
    return {
        'connector_name': settings.name,
        'healthy': settings.healthy,
        'health_endpoint': settings.health_endpoint,
        'health_last_check': settings.health_last_check,
        'health_next_check': settings.health_next_check
    }


@router.get('/ready', tags=['internal'], summary='Readiness check')
def ready() -> dict:
    return {
        'ready': settings.healthy
    }

File: fetch-api/connectors/kube/src/handlers/internal.go
package handlers

import (
	"connector-kube/settings"
	"connector-kube/src/dto/response"

	"github.com/gofiber/fiber/v2"
)


// Health godoc
// @Summary      Health check
// @Tags         health
// @Produce      json
// @Success      200  {object}  response.HealthResponse
// @Router       /api/health [get]
func Health(ctx *fiber.Ctx) error {
	return ctx.JSON(
		response.HealthResponse{
			ConnectorName:  settings.Config.Name,
			Healthy:        settings.Config.Healthy,
			HealthLastCheck: *settings.Config.HealthLastCheck,
			HealthNextCheck: *settings.Config.HealthNextCheck,
		},
	)
}


// Ready godoc
// @Summary      Readiness check
// @Tags         health
// @Produce      json
// @Success      200  {object}  response.ReadyResponse
// @Router       /api/ready [get]
func Ready(ctx *fiber.Ctx) error {
	return ctx.JSON(
		response.ReadyResponse{
			Ready: settings.Config.Healthy,
		},
	)
}
